src/config.rs

```diff
@@ -1,161 +1,16 @@
-        assert_eq!(find_value(content, "key"), Some(String::from("value")));
-    }
-
-    #[test]
-    fn find_value_new_line_at_the_end() {
-        let content = "key=value\n";
-        assert_eq!(find_value(content, "key"), Some(String::from("value")));
-    }
-
-    #[test]
-    fn find_value_trims_value() {
-        let content = "key=value  ";
-        assert_eq!(find_value(content, "key"), Some(String::from("value")));
-    }
-
-    #[test]
-    fn find_value_no_value_after_equals_sign() {
-        let content = "key=";
-        assert_eq!(find_value(content, "key"), None);
-    }
-
-    #[test]
-    fn find_value_whitespace_instead_of_value() {
-        let content = "key= ";
-        assert_eq!(find_value(content, "key"), None);
-    }
-
-    #[test]
-    fn parse_config_from_str_all_fields() {
-        let content = "remote_machine=test@machine\nlocal_compression_level=2\nremote_compression_level=3";
-        assert_eq!(parse_config_from_str(content), Ok(Config {
-            remote_machine_name: String::from("test@machine"),
-            local_compression_level: 2,
-            remote_compression_level: 3,
-        }));
-    }
-
-    #[test]
-    fn parse_config_from_str_all_fields_unordered() {
-        let content = "remote_compression_level=3\nremote_machine=test@machine\nlocal_compression_level=2";
-        assert_eq!(parse_config_from_str(content), Ok(Config {
-            remote_machine_name: String::from("test@machine"),
-            local_compression_level: 2,
-            remote_compression_level: 3,
-        }));
-    }
-
-    #[test]
-    fn parse_config_from_str_only_remote_machine_name() {
-        let content = "remote_machine=test@machine";
-        assert_eq!(parse_config_from_str(content), Ok(Config {
-            remote_machine_name: String::from("test@machine"),
-            local_compression_level: 1, // Default value.
-            remote_compression_level: 1, // Default value.
-        }));
-    }
-
-    #[test]
-    fn parse_config_from_str_no_remote_machine_name() {
-        let content = "local_compression_level=2\nremote_compression_level=3";
-        assert_eq!(parse_config_from_str(content), Err(String::from("please specify 'remote_machine'.")));
-    }
-
-    #[test]
-    fn parse_config_from_str_local_compression_level_not_a_number() {
-        let content = "remote_machine=test@machine\nlocal_compression_level=yooo";
-        assert_eq!(parse_config_from_str(content), Err(String::from("'local_compression_level' must be a positive number, found 'yooo'.")));
-    }
-
-    #[test]
-    fn parse_config_from_str_remote_compression_level_not_a_number() {
-        let content = "remote_machine=test@machine\nremote_compression_level=wut";
-        assert_eq!(parse_config_from_str(content), Err(String::from("'remote_compression_level' must be a positive number, found 'wut'.")));
-    }
+#[derive(Debug, Eq, PartialEq)]
+pub struct Compression {
+    pub local: i64,
+    pub remote: i64,
```

Comment:
Shouldn’t it be `i8` though?

Reply:
yea, it's a slip of YAML impl details, I guess I should use proper type here

Reply:
Not done yet, I'll track it separately, value is validated anyway and realistically i8 is going to still be i32 at least on actual hardware

Reply:
#243